### PR TITLE
at86rf2xx: move PHY and MAC layer components to at86rf2xx_netdev

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
This PR is the first step of #11483 for moving the PHY and MAC logic out of drivers and define common layers for all transceivers.

The changes are:
- Remove `at86rf2xx_get_{pan, long_addr, short_addr}` and add `hwfilter` prefix to the setters. Usually devices are not aware of any kind of addresses (PAN ID, long address, short address) unless there device has hardware address filtering. In that case only setters for hardware address filters are needed (getters might be useful for testing though). In fact, most of this functions are replaced by netdev_ieee802154 stuff.
- Merge `at86rf2xx_set_channel` and `at86rf2xx_set_page` into `at86rf2xx_configure_phy`. In practice, the tuple `(channel_number, channel_page)` describes the PHY channel configuration, and these 2 numbers are not independent.
- Change API of power and rx sensitivity getters/setters: I noticed they always convert from dBm to and from the internal representation values. In practice, upper layers are only interested in the value in dBm. Thus, this logic is handled in the netdev abstraction and the device only sets/gets the internal value.
- Move TX start event to netdev.

With this change, the driver is mostly independent of `netdev`.

### Testing procedure
Run `tests/drivers_at86rf2xx` and perform common operations (change addresses, send packets, etc).

### Issues/PRs references
~~Depends on #11718~~
#11483

### CODE REVIEW

**Comment by kaspar030** [comment]
>     * Change API of power and rx sensitivity getters/setters: I noticed they always convert from dBm to and from the internal representation values. In practice, upper layers are only interested in the value in dBm. Thus, this logic is handled in the netdev abstraction and the device only sets/gets the internal value.

This I don't get. The device obviously need to set the internal values, but it is also the only part that knows how to convert between internal values and dbm. Doesn't it need to expose dbm API to any upper layer?

**Comment by jia200x** [comment]
> This I don't get. The device obviously need to set the internal values, but it is also the only part that knows how to convert between internal values and dbm. Doesn't it need to expose dbm API to any upper layer?

In practice yes, there should be an API for the upper layers. Since `netdev` is the only layer on top so far, I didn't add these functions. We might need them at some point anyway.

Also, I'm only considering `dBm -> internal value` since the `txpower` and `rxsensitivity` is usually reported via the PHY PIB and not from the SPI registers. Not sure if there might be a use case for the `internal value -> dBm` but I could add it anyway.

File: drivers/at86rf2xx/at86rf2xx_internal.c [hunk 1]
```diff
@@ -149,16 +149,15 @@ void at86rf2xx_hardware_reset(at86rf2xx_t *dev)
              && (dev->state != AT86RF2XX_STATE_P_ON));
 }
 
-void at86rf2xx_configure_phy(at86rf2xx_t *dev)
+void at86rf2xx_configure_phy(at86rf2xx_t *dev, uint8_t chan, uint8_t page)
```
[the remark this answers is not in the document]
Comment (jia200x) [reply drivers/at86rf2xx/at86rf2xx_internal.c]: sure, considering the fact there are other PHY variables which are not being set (txpower, etc). I will ;)

**Comment by kaspar030** [comment]
> In practice yes, there should be an API for the upper layers. Since `netdev` is the only layer on top so far, I didn't add these functions.

Well, ```ifconfig``` displays dbm values. Internal (register scale) values are not really interesting. ```netdev_ieee802154``` doesn't know any of them. If the driver <-> 802154 API is using internal values, 802154 can only get and then set again an internal values with no means of presenting them to the user, or use any kind device agnostic configuration. Thus I'm strongly for using dbm values for device <-> netdev_802154.

>  We might need them at some point anyway.

The conversion code is already there. Maybe we shouldn't remove it now just to add it as soon as we realize the rework doesn't allow displaying/setting dbm values anymore.

**Comment by kaspar030** [comment]
I'm not entirely happy to see the "PHY" and "MAC" terminology here. Traditionally they refer to actual hardware ICs, which doesn't apply here. "MAC" as in "handling the logical packet format" also doesn't fit. We will have netdev_802154 taking care of that, with another layer (that is replaceable) using its API to do actual access control.

**Comment by jia200x** [comment]
> I'm not entirely happy to see the "PHY" and "MAC" terminology here. Traditionally they refer to actual hardware ICs, which doesn't apply here. 

Hmmm when talking about PHY and MAC I'm refering to the definitions of IEEE802.15.4 MAC and PHY. Both components have their own interfaces and duties (see e.g [here](http://www.ismlab.usf.edu/dcom/Ch8_802.15.4-2006.pdf), section 6.2)

 > "MAC" as in "handling the logical packet format" also doesn't fit. We will have netdev_802154 taking care of that, with another layer (that is replaceable) using its API to do actual access control.

I agree. The thing is, stuff like PAN ID and addresses are pure MAC terminology, so at least they shouldn't be present in the driver (all .c files except at86rf2xx_netdev.c).

I think the title is misleading here. I'm not moving the PHY and MAC to the actual netdev interface, but I'm just removing the PHY and MAC component from the main driver implementation and put it for now in at86rf2xx_netdev.c

**Comment by jia200x** [comment]
@kaspar030 I changed the title to reflect the proposed change

**Comment by benpicco** [comment]
I guess now that #11718 is merged, this needs a rebase.

**Comment by jia200x** [comment]
this driver has changed a lot since I opened this PR, so I will close it for now.

**Pull request closed without merging**